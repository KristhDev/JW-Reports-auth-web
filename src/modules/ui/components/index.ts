export { Button } from './Button';
export { EyeBtn } from './EyeBtn';
export { FormField } from './FormField';
export { Spinner } from './Spinner';
export { Title } from './Title';