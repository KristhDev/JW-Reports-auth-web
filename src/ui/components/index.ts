export { FormField } from './FormField';
export { Title } from './Title';