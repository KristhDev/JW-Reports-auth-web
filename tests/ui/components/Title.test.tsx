import { render, screen } from '@testing-library/react';

import { Title } from '../../../src/ui/components';

const titleText = 'Test title';

describe('Test in <Title /> component', () => {
    beforeEach(() => {
        render(
            <Title
                classNameContainer="container-test"
                classNameText="text-test"
                text={ titleText }
            />
        );
    });

    it('should to match the snapshot', () => {
        const { container } =  render(
            <Title text={ titleText } />
        );

        expect(container).toMatchSnapshot();
    });

    it('should render respective props', () => {
        const containerElement = screen.getByTestId('title-container');
        const textElement = screen.getByRole('heading');

        expect(containerElement.className).toContain('container-test');
        expect(textElement.className).toContain('text-test');
        expect(textElement.innerHTML).toBe(titleText);
    });
});