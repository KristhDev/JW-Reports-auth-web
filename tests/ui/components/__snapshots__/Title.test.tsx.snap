// Vitest Snapshot v1

exports[`Test in <Title /> component > should to match the snapshot 1`] = `
<div>
  <div
    class="title-container undefined"
    data-testid="title-container"
  >
    <h1
      class="title-text undefined"
    >
      Test title
    </h1>
  </div>
</div>
`;
